files/generate_system_metrics

```diff
@@ -0,0 +1,253 @@
+      sar_output_arr = sar_output.split(/\n+|\r+/).reject(&:empty?).map { |line| line.split }
+
+      if ( @metric_type == "memory")
+        unique_header_str = "%memused"
+      else
+        unique_header_str = "%user"
+      end
+      headers_line = sar_output_arr.find { |e| e.include? unique_header_str }
+
+      sar_error_missing_headers = "sar output invalid or missing headers." +
+                                  "failed to find line with #{unique_header_str}."
+                                  "\nFull output:\n#{sar_output}"
+      raise(sar_error_missing_headers) if headers_line.nil?
+
+      averages_line = sar_output_arr.find { |e| e.include? "Average:" }
+      sar_error_missing_averages = "sar output missing \"Average:\"\nFull output:\n#{sar_output}"
+      raise(sar_error_missing_averages) if averages_line.nil?
+
+      Hash[headers_line.reverse.zip(averages_line.reverse).reverse]
+
+      puts "sar headers and averages:\n#{headers_line.join(",")}\n#{averages_line.join(",")}" if @verbose
+
+      #example of array data
+      # 04:59:13,PM,CPU,%user,%nice,%system,%iowait,%steal,%idle
+      # Average:,all,0.58,0.00,0.08,0.00,0.00,99.33
+      #combine the arrays into a hash starting from the deal with the unmatched columns in the front
+      data_hash = Hash[headers_line.reverse.zip(averages_line.reverse).reverse]
+      # remove anything that doesn't have a number for an average like "Average:" or "all"
+      data_hash.select{ |k,v| v =~ /\A[-+]?[0-9]*\.?[0-9]+\Z/ }
+    end
+
+    # Convert the inputted sar output into json and raise errors if the sar output is invalid
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] sar_output
+    # @param [obj] time_stamp_obj Time object to use for generating the filename
+    #
+    # @raise [RuntimeError] if sar_output doesn't parse correctly
+    #
+    # @return [string] json of sar output
+    def convert_sar_output_to_json(sar_output, time_stamp_obj)
+      hostkey = @hostname.gsub('.', '-')
+      dataset = {'time_stamp_obj' => time_stamp_obj.utc.iso8601, 'servers' => {}}
+      metrics_data = parse_sar_output(sar_output)
+      dataset['servers'][hostkey] = {@metric_type => metrics_data}
+      json_dataset = JSON.pretty_generate(dataset)
+      return json_dataset
+    end
+
+    # Create the file and put the json data in it
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] json_dataset data in json format to put in file
+    # @param [obj] time_stamp_obj Time object to use for generating the filename
+    #
+    # @return [void]
+    def create_file(json_dataset, time_stamp_obj)
+      filename = time_stamp_obj.utc.strftime('%Y%m%dT%H%M%SZ') + '.json'
+      dirname = "#{@metrics_dir}/system_#{@metric_type}/#{@hostname}"
+      file_path = "#{dirname}/#{filename}"
+      FileUtils.mkdir_p(dirname) unless File.directory?(dirname)
+      puts "Creating json file: #{file_path}" if @verbose
+      File.open(file_path, 'w') do |file|
```

Comment:
Would the 'short version' be better here or is it better to use the longer form?
```
File.write(file_path, json_dataset)
```